packages/source/types/data.ts

```diff
@@ -1,379 +1,395 @@
-    is504?: boolean;
-    is505?: boolean;
-  }
->;
+  // This is ugly but it seems like the best way.
+  // Also types are erased at runtime so it doesnt add to bundle size
 
-// ARCHIVES
+  /* eslint-disable jsdoc/require-jsdoc */
+  is400?: boolean;
+  is401?: boolean;
+  is402?: boolean;
+  is403?: boolean;
+  is404?: boolean;
+  is405?: boolean;
+  is406?: boolean;
+  is407?: boolean;
+  is408?: boolean;
+  is409?: boolean;
+  is410?: boolean;
+  is411?: boolean;
+  is412?: boolean;
+  is413?: boolean;
+  is414?: boolean;
+  is415?: boolean;
+  is416?: boolean;
+  is417?: boolean;
+  is500?: boolean;
+  is501?: boolean;
+  is502?: boolean;
+  is503?: boolean;
+  is504?: boolean;
+  is505?: boolean;
+  /* eslint-enable */
+}
 
-/**
- * Adds properties to `BaseData` to identify archive pages.
- *
- * @property {true} isArchive
- * @property {EntityData[]} items - List of items contained in this archive page.
- * @property {number} page - The page number of this archive page.
- * @property {string} path - The path of this archive page without the page.
- * @property {string} next - The link to the next page if it exists.
- * @property {string} previous - The link to the previous page if it exists.
- * @property {number} total - Total number of post entities in the whole archive.
- * @property {number} total - Total number of pages in the whole archive.
- * @property {false} isSearch
- */
-export type ArchiveData = Merge<
-  BaseData,
-  {
-    isArchive: true;
-    items: EntityData[];
-    page?: number;
-    path?: string;
-    next?: string;
-    previous?: string;
-    total?: number;
-    totalPages?: number;
-  }
->;
+// ARCHIVES
 
 /**
- * Adds properties to identify archive pages with search.
+ * Data that references a post type entity in the state.
  *
- * @property {true} isSearch
- * @property {string} searchQuery
+ * This kind of objects are included in an array called `items` present in
+ * {@link ArchiveData} objects.
  */
-export type SearchData = {
-  isSearch: true;
-  searchQuery: string;
-};
+export interface DataItem {
+  /**
+   * Entity post type.
+   */
+  type: string;
 
-/**
- * Adds properties to `ArchiveData` to identify taxonomy pages.
- *
- * @property {true} isTaxonomy
- * @property {string} taxonomy - Taxonomy slug.
- * @property {number} id - Taxonomy id.
- */
-export type TaxonomyData = Merge<
-  ArchiveData,
-  {
-    isTaxonomy: true;
-    taxonomy: string;
-    id: number;
-  }
->;
+  /**
+   * Entity ID.
+   */
+  id: number;
 
-/**
- * Adds properties to `TaxonomyData` to identify taxonomy with search pages.
- */
-export type TaxonomyWithSearchData = Merge<TaxonomyData, SearchData>;
+  /**
+   * Link where this entity is shown.
+   */
+  link: string;
+}
 
 /**
- * Adds properties to `TaxonomyData` to identify category pages.
+ * Data that references a post type entity in the state.
  *
- * @property {true} isCategory
- * @property {"category"} taxonomy
+ * @deprecated Use {@link DataItem} instead.
```

Comment:
Makes sense, `items` is definitely not the best name. :sweat_smile: 

Regarding `data.entities`, we should make clearer that the main entity would still be referenced with `data.type`, `data.id`, or other props, and `data.entities` would be only for the listed entities (usually post type entities).

Reply:
Oh, yes, `data.entities` would only be used to replace `data.items`.

Reply:
Should I rename it to `DataEntity` then, or something similar? Just to make a decision here.

Reply:
Yes, something like that 🙂

Reply:
In my opinion, we can use `DataItem` now and change it later to `DataEntity` - or whatever - once we change `data.items` to `data.entities`. What do you think, @luisherranz?

Reply:
My plan is to make the `data.items` -> `data.entities` transition a backward-compatible change using Frontity hooks for the derived state. So knowing that, I think we should start using the `DataEntity` name now to avoid breaking people's types in the future.

Reply:
Gotcha. I'll change that as well.